Classes/Domain/Dto/CurrentPage.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WebVision\Deepltranslate\Core\Domain\Dto;
+
+use WebVision\Deepltranslate\Core\Exception\MissingParameterException;
+
+final class CurrentPage
+{
+    public function __construct(
+        private readonly int $uid,
```

Comment:
Public constructor property promotion can be used here easily, eliminates the need for the getters.

Additionally, it would follow the TYPO3 Core recommendation for using naming arguments as a generic concept here. Other DTO's may have optional stuff with defaults and public CPP may then be really be usefull.

https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/CodingGuidelines/CodingBestPractices/NamedArguments.html#utilizing-named-arguments-in-extensions